instrumentation/github.com/Shopify/sarama/otelsarama/metric.go

```diff
@@ -0,0 +1,98 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelsarama // import "go.opentelemetry.io/contrib/instrumentation/github.com/Shopify/sarama/otelsarama"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/asyncfloat64"
+	"go.opentelemetry.io/otel/metric/unit"
+	"go.uber.org/atomic"
+)
+
+const (
+	saramaKafkaVersion = 2
+)
+
+type rateMetric struct {
+	startedAt          time.Time
+	recordAccumulation atomic.Float32
+}
+
+// rate metric to be used for calculation of per second average
+func NewRateMetric() rateMetric {
+	return rateMetric{
+		startedAt:          time.Now(),
+		recordAccumulation: *atomic.NewFloat32(0),
+	}
+}
+
+func (m *rateMetric) Add(record float64) {
+	m.recordAccumulation.Add(float32(record))
+}
+
+func (m *rateMetric) Average() float64 {
+	secondElapsed := time.Since(m.startedAt).Seconds()
+
+	var res float64 = float64(m.recordAccumulation.Load())
+	if secondElapsed > 1 {
+		res = res / secondElapsed
+	}
+	m.flush()
+	return res
+}
+
+func (m *rateMetric) flush() {
+	m.startedAt = time.Now()
+	m.recordAccumulation.Swap(0)
```

Comment:
`flush` is only called here. It seems like a good idea to merge `flush` into `Average` and have `Average` be the only method that works with `startedAt` to avoid a future race condition.